pkg/querier/tenantfederation/tenant_federation.go

```diff
@@ -0,0 +1,14 @@
+package tenantfederation
+
+import (
+	"flag"
+)
+
+type Config struct {
+	// Enabled switches on support for multi tenant query federation
+	Enabled bool `yaml:"enabled"`
+}
+
+func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
+	f.BoolVar(&cfg.Enabled, "tenant-federation.enabled", false, "If enabled on all components, queries can be federated across multiple tenants. The tenant IDs involved need to be specified separated by a `|` character in the `X-Scope-OrgID` header (experimental).")
```

Comment:
Let's also mention experimental feature in the `v1-guarantees.md` file.

Reply:
This is done in 891e692f79a1a65fe6a33961b6bb8f13121a5253